core/src/rpc.rs

```diff
@@ -35,6 +35,32 @@ use std::{
     time::{Duration, Instant},
 };
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RpcResponseContext {
+    pub slot: u64,
+    pub block: String,
+    pub leader: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Response<T> {
+    pub context: RpcResponseContext,
+    pub value: Result<T>,
```

Comment:
I don't think we should wrap a Result around T here by default

Reply:
OK cool, forgive me, some signatures might look more verbose as that `Result<T>` goes back out into the world